src/app/utils/socket-utils.ts

```diff
@@ -458,9 +461,47 @@ export function getWeaponSockets(
     PlugCategoryHashes.V300WeaponDamageTypeKinetic,
   ];
 
+  let moddedSockets: DimSockets = item.sockets;
+  if (includeFakeMasterwork) {
+    const allSockets = item.sockets.allSockets.map((socket) => {
+      if (socket.socketDefinition.socketTypeHash !== weaponMasterworkY2SocketTypeHash) {
+        return socket;
+      }
+      const mwHash = item.masterworkInfo?.stats?.find((s) => s.isPrimary)?.hash || 0;
+      const plugCategory = D2PlugCategoryByStatHash.get(mwHash);
+      let fullMasterworkPlug =
+        socket.plugSet &&
+        plugCategory &&
+        maxBy(
+          socket.plugSet.plugs.filter((p) => p.plugDef.plug.plugCategoryHash === plugCategory),
+          (plugOption) => plugOption.plugDef.investmentStats[0]?.value,
+        );
+      if (!fullMasterworkPlug) {
+        return socket;
+      }
+      fullMasterworkPlug = {
+        ...fullMasterworkPlug,
+        plugDef: { ...fullMasterworkPlug.plugDef, iconWatermark: '', investmentStats: [] },
```

Comment:
Why do you need to overwrite the watermark and investment stats?

Reply:
Without overwriting watermark we get a `10` in the top left corner of the icon.
Without overwriting stats we have the potential for the stats on the fake plug to actually affect the stats on the weapon, somewhere along the line in our parsing.